NeuroID/src/main/java/com/neuroid/tracker/NeuroIDPublic.kt

```diff
@@ -0,0 +1,189 @@
+package com.neuroid.tracker
+
+import android.app.Activity
+import androidx.annotation.VisibleForTesting
+import com.neuroid.tracker.models.SessionStartResult
+
+interface NeuroIDPublic {
+    /**
+     * Deprecated, do not use!
+     */
+    @Deprecated("Replaced with getUserID", ReplaceWith("getUserID()"))
+    fun getClientId():String
+
+    /**
+     * Return the currently set client id (use this one).
+     */
+    fun getClientID():String
+
+    /**
+     * Set a tag for the widget that is not registered.
+     */
+    fun setTestURL(newEndpoint: String)
+
+    /**
+     * Set a dev testing URL for testing.
+     */
+    @VisibleForTesting
+    fun setTestingNeuroIDDevURL()
+
+    /**
+     * Set a user id.
+     */
+    fun setUserID(userID: String): Boolean
+
+    /**
+     * Return the currently set user id.
+     */
+    fun getUserId(): String
+    /**
+     * Return the currently set user id.
+     */
+    fun getUserID(): String
+    /**
+     * Return the currently set registered user id.
+     */
+    fun getRegisteredUserID(): String
+
+    /**
+     * Set a registered user id.
+     */
+    fun setRegisteredUserID(registeredUserId: String): Boolean
+
+    /**
+     * Set a screen name for the current page in a session.
+     */
+    fun setScreenName(screen: String): Boolean
+
+    /**
+     * Return the currently set screen name for the current page in a session.
+     */
+    fun getScreenName(): String
+
+    /**
+     * Register page targets. This will send events targets to the server for the current page.
+     */
+    fun registerPageTargets(activity: Activity)
```

Comment:
Should we note this is only for RN? I don't think an Android user would have reason to call this

Reply:
Good point. updated